Content.Server/CartridgeLoader/CartridgeLoaderSystem.cs

```diff
@@ -191,7 +191,10 @@ public bool InstallProgram(EntityUid loaderUid, string prototype, bool deinstall
         if (container.Count >= loader.DiskSpace)
             return false;
 
-        // TODO cancel duplicate program installations
+        foreach (var program in GetInstalled(loaderUid))
+            if (Prototype(program)?.ID == prototype)
+                return false;
+
```

Comment:
You can't use entityprototype checks to compare whether entities are equals.

As above:

// For anyone stumbling onto this: Do not do this or I will cut you.

Reply:
im not sure i fully understand, what problems would it cause?